.github/workflows/submit-github-dependency-graph.yml

```diff
@@ -0,0 +1,26 @@
+name: Submit GitHub Dependency Graph
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  
+permissions: write-all
+
+jobs:
+  generate-and-submit-graph:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-java@v3
+      with:
+        distribution: temurin
+        java-version: 8
+    - name: Setup Gradle
+      uses: gradle/gradle-build-action@v2    
+    - name: Generate dependency graph
+      uses: gradle/gradle-build-action/actions/dependency-graph-generate@v2
+      env:
+        GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_SOLUTIONS_ACCESS_TOKEN }}
+    - name: Submit dependency graph
```

Comment:
Just curious if you have considered merging the generate and submit steps?
is this for troubleshooting reasons or that other sinks could be used?

Reply:
The main reason for the separation is that it's necessary for PR builds, where the "submit" step has to be done in a separate workflow with sufficient permissions.